datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -991,6 +991,9 @@ fn binary_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<DataType>
         (Binary | Utf8, Binary) | (Binary, Utf8) => Some(Binary),
         (LargeBinary | Binary | Utf8 | LargeUtf8, LargeBinary)
         | (LargeBinary, Binary | Utf8 | LargeUtf8) => Some(LargeBinary),
+        (BinaryView, BinaryView) | (BinaryView, Binary) | (Binary, BinaryView) => {
```

Comment:
🤔  I wonder if we should also handle `LargeBinary` as well

Reply:
We also don't handle `LargeUtf8`. :eyes:

Reply:
Good call -- I filed https://github.com/apache/datafusion/issues/11023 to track